extensions/cms/src/cmsUtils.ts

```diff
@@ -79,40 +105,65 @@ export class CmsUtils {
 		});
 	}
 
+	public static didConnectionChange(connectionA: azdata.connection.Connection, connectionB: azdata.connection.Connection): boolean {
+		return (connectionA !== connectionB) || ((connectionA.connectionId === connectionB.connectionId) &&
+			(connectionA.options.savePassword !== connectionA.options.savePassword));
+	}
+
 	public async createCmsServer(connection: azdata.connection.Connection,
-		name: string, description: string): Promise<mssql.ListRegisteredServersResult> {
+		name: string, description: string): Promise<CreateCmsResult> {
 		let provider = await this.getCmsService();
 		connection.providerName = connection.providerName === cmsProvider ? mssqlProvider : connection.providerName;
 		let ownerUri = await azdata.connection.getUriForConnection(connection.connectionId);
 		if (!ownerUri) {
 			// Make a connection if it's not already connected
-			await azdata.connection.connect(Utils.toConnectionProfile(connection), false, false).then(async (result) => {
+			let initialConnectionProfile = CmsUtils.getConnectionProfile(connection);
+			await azdata.connection.connect(initialConnectionProfile, false, false).then(async (result) => {
 				ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
```

Comment:
Remove the .then - you're already going to be waiting because of the await anyways.

Reply:
In general we're trying to encourage use of async/await over the old Promise/Thenable/then. async/await is a LOT easier to read and less error prone since you don't have to parse through nested function calls all the time.